### PR TITLE
Bugfix: Delete old Zendesk tickets

### PR DESCRIPTION
**Pull request by TobyRet**
- Fix active job bug (do not pass in the Zendesk client)
- Write tests for Zendesk client for code coverage

### CODE REVIEW

**Comment by StupidCodeFactory** [comment]
@TobyRet can you explain what the bug was?

File: app/jobs/zendesk_tickets_job.rb [hunk 1]
```diff
@@ -19,8 +20,8 @@ def perform(client, feedback)
   # the service field is 'prison_visits' and another one for staff that matches
   # tickets tagged with 'staff.prison.visits'.
 
-  def ticket_raised!(client, feedback)
-    ZendeskAPI::Ticket.create!(client, ticket_attrs(feedback))
+  def ticket_raised!(feedback)
+    ZendeskAPI::Ticket.create!(self.class.client, ticket_attrs(feedback))
```
Comment (StupidCodeFactory) [inline app/jobs/zendesk_tickets_job.rb]: I'm really not convinced about the benefit of the client being a class level property.
What happens if two jobs tries to make a request to zendesk at the same time? :trollface: 

We can ameliorate this later though

Comment (TobyRet) [reply app/jobs/zendesk_tickets_job.rb]: @StupidCodeFactory - yes this a bit shit but you cannot add instance methods to classes inheriting from ActiveJob. Class methods are possible though. The alternative is to build the client and not use the module ... Happy to do this if you prefer, but then there is no need for a module ...

Comment (StupidCodeFactory) [reply app/jobs/zendesk_tickets_job.rb]: Wait whaaaaaaat? one can't add instance method to ActiveJob class?? What's the rational can you point me where this is stated?

File: spec/services/zendesk_client_spec.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe ZendeskClient do
+  subject { Class.new }
+
+  URL = 'https://zendesk_api.com'
+  USERNAME = 'bob'
+  TOKEN = '123456'
```
Comment (StupidCodeFactory) [inline spec/services/zendesk_client_spec.rb]: these should `let`

File: spec/services/zendesk_client_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  expectations = [
```
Comment (StupidCodeFactory) [inline spec/services/zendesk_client_spec.rb]: same for this statment

**Pull request merged**